### PR TITLE
[HttpKernel] Allow disabling write permission check for log dir

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #16572 |
| License | MIT |
| Doc PR |  |

Continuing #16583

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Exception/MethodNotImplementedException.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Exception;
+
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
```
Comment (unkind) [inline src/Symfony/Component/HttpKernel/Exception/MethodNotImplementedException.php]: Wrong exception class.

Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Exception/MethodNotImplementedException.php]: better now?

**Comment by fabpot** [comment]
I don't think this is worth it. If someone does not want the log dir, they can just point it to something under the cache dir, and be done with it. As the use case is really rare, I don't think we need to do anything in Symfony.

**Comment by nicolas-grekas** [comment]
I agree, closing then

**Comment by nesl247** [comment]
Can this be discussed again? One of the recommendations for security in containerized environments is to have a read only filesystem. This means that there is no directory that should actually be writeable unless necessary. When using `stdout` as is the common case with containers, the log dir is no longer necessary to exist. While the previous answer of pointing it to something under the cache dir is certainly an option, it's not the best user experience.

**Comment by erinarbleizez** [comment]
I confirm we have that use case in our company with 2 clients wanting there front server in paranoid/read only for web server's user (and for some reason I won't explain here for security reason, we need that symfony copy on the front server, not just a varnish read only cache).

For cache we can still cache:warmup with the bin/console but the log directory still needs write access for web user. being able to point it at least to stdout/stderr instead of the disk drive would be great. Being able to completely disable logs on prod would be better.

**Pull request closed without merging**